polkadot/node/core/pvf/src/artifacts.rs

```diff
@@ -258,7 +221,13 @@ impl Artifacts {
 					return true
 				},
 			};
-			!file_name.ends_with(checksum.to_hex().as_str())
+
+			if let Some(file_name) = path.file_name {
+				if let Some(file_name) = file_name.to_str() {
+					return !file_name.ends_with(checksum.to_hex().as_str())
+				}
+			}
+			false
```

Comment:
Should be `true` right? Could use a test for this case if we don't have one yet.

Reply:
Oops! That's right. But this line of can only be reached when

* the path end in "/..", or
* the path contain non-Unicode

These are all screened out at the call site, so I don't even know how to write a test that could fail because of this line lol.

Reply:
I guess `is_corrupt` can be extracted and unit-tested, but whatever, looks good already!

Reply:
We should probably stick to testing the interfaces, this kind of tests often become maintenance liability (prone to change). 😉